### DISCUSSION TITLE
is it possible to restore a (newer!) local file version overwritten by a onedrive sync?

### DISCUSSION THREAD

**Discussion by m4chi** [abraunegg/onedrive#1813] [Q&A] [answered]
Dear community, I have a big problem. I have worked on document file for a while and am pretty sure I have saved it several times on the way, over multiple days. A recent sync with the onedrive client has apparently overwritten the changes, restoring a previous, much older version of the file. When I opened Libre Office, I could still see the new version in the small preview window, but opening the file leads me to the older, original version. Since I have spent several days working on this, I was wondering if anyone has a suggestion for restoring the newer file version. I checked the manual sync I did and it simply states "downloading file..." even though the file version in the cloud is much older. There is no file history on the cloud, because now the older version is the only one I have.
I am working in Linux Ubuntu 18.04, unfortunately without any logging for my onedrive syncs (didn't have any issues so far). I also have to admit that I am not an expert of Unix file systems. Any help would be much appreciated.
I think the same happened in discussion #1359 . For additional info, my client version is onedrive v2.4.10-11-g7786a17
I know this is not the most recent version, but in my defense, i only used that computer as a temporary fallback while it was difficult to get to the office...

**Comment by abraunegg** [comment, marked as answer]
@m4chi 
> A recent sync with the onedrive client has apparently overwritten the changes, restoring a previous, much older version of the file. 

The default configuration of the client, when this happens - is to rename the conficting file - so you have a backup of the local file already.

If you have however enabled the following in your configuration file:
```
bypass_data_preservation = "true"
```

Then no backup will be made.

If you have *any* other OneDrive client version installed (such as the 'skilion' client) - that client has a history of causing data-loss. Check your entire system for 'onedrive' binaries.

> I am working in Linux Ubuntu 18.04,

Do not use Ubuntu 18.x. Upgrade to a more modern / more current version | distribution.

> For additional info, my client version is onedrive v2.4.10-11-g7786a17

That is obsolete and unsupported. You must monitor the releases and rebuild your client on a regular basis, or, upgrade your Ubuntu version to a more modern version and use the OpenSuSE Build Service to install the client from supported packages that will then keep your client current and supported.

**Reply by m4chi** [reply]
@abraunegg 
thank you very much for the prompt advise, much appreciated! I have not set the bypass_data_preservation to true, but there is no renamed version of the file to be found anywhere. I know I should update the system, but now that things have gone wrong, I was hoping for some chance to get my file back. Or, if there is no chance of getting the file back, at least I know the work is lost...

**Reply by abraunegg** [reply]
@m4chi 
> but there is no renamed version of the file to be found anywhere 

Unfortunately if there is no backup file in the format of `<filename>-<hostname>.<extention>`, the required local version is gone.

So why could that be - when, the application 'default' is to make a backup of the conflicting file? Some plausible items:

1. The application made the request, your file system tried to copy / rename the file but encountered an error - thus the rename failed
2. You ran the client with a number of enabled options that triggered an obscure bug / issue
3. You ran the client with `--resync` and due to the above point this triggered a different obscure bug / issue.
4. You encountered a filesystem issue which for some reason meant that the items.sqlite3 file - even though it was present - could not be found / accessed during client operations. This was bug https://github.com/abraunegg/onedrive/issues/1714 and was fixed in v2.4.14. You are using a much older client version that is almost 12 months old now.

I absolutely have zero acceptance for data-loss. I have gone to great efforts to ensure the client is as bug-free as possible &, when conflicts occur to ensure that a backup is made. If you care about your data & your data integrity:

1. Do not use any local filesystem type other than ZFS. Educate yourself as to why. Enable ZFS filesystem snapshots at regular intervals for your document | data volumes that you care about so that you can 'roll back' if required to restore or recover data quickly.
2. OneDrive is not a 'backup' solution. Ensure you have a external disk / external system where you copy your data to that external device on a regular backup on a regular (minimum daily) basis.
3. Ensure your OS is kept up-to-date. Unfortunately Ubuntu is a quagmire mess when it comes to keeping it 'current' due to its package policies - so maybe look at other, better maintained distributions like Alpine, Arch|Manjaro, Fedora, Debian, OpenSuSE, Slackware and Solus. 
4. Ensure that you use a Linux distribution that keeps the 'onedrive' package current. If you are building from source the onus is on you to ensure you monitor the releases on GitHub and rebuild the client when a new release occurs.
5. When running | using the client - only use `--resync` if the client tells you to. Otherwise - do not use that option as it is meant as a 'last resort item'. This really should be made clearer in the documentation and/or even have a 'are you sure' type of confirmation when used.
6. Enable application logging so that anytime you run the application, you have a log file of what the application actually did
